### ISSUE TITLE
(Question) Why modifying the BLOCK size will cause errors?

### ISSUE THREAD

**Issue by wuliJerry** [HazyResearch/flash-attention#221]
Hi there!
I am trying to use the [triton version](https://github.com/HazyResearch/flash-attention/blob/main/flash_attn/flash_attn_triton.py) to run some experiments, by modifying the `BLOCK` in the [`_flash_attn_forward` function](https://github.com/HazyResearch/flash-attention/blob/eff9fe6b8076df59d64d7a3f464696738a3c7c24/flash_attn/flash_attn_triton.py#L620) into different values. 
However, I encountered some errors when the `BLOCK` is changed to 256, 64, 32, and other values, the specific log is attached [error_log.txt](https://github.com/HazyResearch/flash-attention/files/11468718/error_log.txt) (I also modified the [auto-tune config](https://github.com/HazyResearch/flash-attention/blob/eff9fe6b8076df59d64d7a3f464696738a3c7c24/flash_attn/flash_attn_triton.py#L486) into the respective value). The triton version I used is 20221202dev as well.
From the code, I thought we only need to make sure that `seqlen_k % BLOCK == 0` and `seqlen_q % BLOCK == 0`,  so these errors don't make any sense to me.

**Comment by wuliJerry** [comment]
Also, I only tried the forward method in `FlashAttnFunc` for the experiment.

**Comment by tridao** [comment]
Yeah the Triton errors don't make sense to me either, maybe because I don't understand the details of the Triton compiler. The [implementation](https://github.com/openai/triton/blob/47af6ba702a99baebadfbaf5ecab8c71ce360562/python/triton/ops/flash_attention.py#L207) from Triton repo also only works for headdim = 64 (and not headdim=128) right now.
